hardware_interface/doc/joints_userdoc.rst

```diff
@@ -0,0 +1,132 @@
+
+Serial Kinematics
+  Serial kinematics refers to the arrangement of joints in a robotic manipulator where each joint is independent of the others, and the number of joints is equal to the DOF of the kinematic chain.
+
+  A typical example is an industrial robot with six revolute joints, having six DOF. Each joint can be actuated independently, and the end-effector can be moved to any position and orientation in the workspace.
+
+Kinematic Loops
+  On the other hand, kinematic loops, also known as closed-loop mechanisms, involve several joints that are connected in a kinematic chain and being actuated together. This means that the joints are coupled and cannot be moved independently: In general, the number of DOFs is smaller than the number of joints.
+  This structure is typical for parallel kinematic mechanisms, where the end-effector is connected to the base by several kinematic chains.
+
+  An example is the four-bar linkage, which consists of four links and four joints. It can have one or two actuators and consequently one or two DOFs, despite having four joints. Furthermore, we can say that we have one (two) actuated joint and three (two) passive joints, which must satisfy the kinematic constraints of the mechanism.
+
+URDF
+#############
+
+URDF is the default format to describe robot kinematics in ROS. However, only serial kinematic chains are supported, except for the so-called mimic joints.
+See the `URDF specification <http://wiki.ros.org/urdf/XML/joint>`__ for more details. For the mimic joints, we can read
+
+  This tag is used to specify that the defined joint mimics another existing joint. The value of this joint can be computed as value = multiplier * other_joint_value + offset.
+
+It can be defined in the following way in the URDF
+
+  .. code-block:: xml
+
+    <joint name="right_finger_joint" type="prismatic">
+      <axis xyz="0 1 0"/>
+      <origin xyz="0.0 -0.48 1" rpy="0.0 0.0 0.0"/>
+      <parent link="base"/>
+      <child link="finger_right"/>
+      <limit effort="1000.0" lower="0" upper="0.38" velocity="10"/>
+    </joint>
+    <joint name="left_finger_joint" type="prismatic">
+      <mimic joint="right_finger_joint" multiplier="1" offset="0"/>
+      <axis xyz="0 1 0"/>
+      <origin xyz="0.0 0.48 1" rpy="0.0 0.0 3.1415926535"/>
+      <parent link="base"/>
+      <child link="finger_left"/>
+      <limit effort="1000.0" lower="0" upper="0.38" velocity="10"/>
+    </joint>
+
+Mimic joints are an abstraction of the real world: They can be used to describe
+
+* simple parallel kinematics with linear dependencies of the joint positions
+* abstract complex groups of actuated joints, where the positions of several joints are directly controlled by low-level control loops and move synchronously.
```

Comment:
I don't have one, I just can imagine that this could be a usecase ;) Or wheels of a multi-wheel diffbot, or fans of a drone with rotating at the same speed...

Reply:
Maybe leave some examples here, then it would be clearer. The belt and pulley system could also be a  example?

Reply:
what do you think about the last update?